packages/public/server/src/module/Frontend/src/Frontend/Factory/RenderComponentHelperFactory.php

```diff
@@ -22,22 +22,19 @@
  */
 namespace Frontend\Factory;
 
+use Frontend\RenderComponentService;
 use Frontend\View\Helper\RenderComponentHelper;
 use Zend\ServiceManager\FactoryInterface;
 use Zend\ServiceManager\ServiceLocatorInterface;
 
 class RenderComponentHelperFactory implements FactoryInterface
 {
-    /**
-     * Create service
-     *
-     * @param ServiceLocatorInterface $serviceLocator
-     * @return mixed
-     */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
+        /* @var ServiceLocatorInterface $serviceLocator */
         $serviceLocator = $serviceLocator->getServiceLocator();
-        $url   = $serviceLocator->get('config')['services']['frontend'];
-        return new RenderComponentHelper($url);
+        /* @var RenderComponentService $service */
+        $service = $serviceLocator->get(RenderComponentService::class);
```

Comment:
Extracted the logic into its own service since `$serviceLocator` behave differently in view helpers vs. service manager. So we should only use view helpers in the view and use the service in controllers / other services / etc. (cc @Knorrke since you had a similar issue / question a couple of weeks ago)